doc/contributing.md

```diff
@@ -192,8 +192,16 @@ It will make your life a lot easier!
 
 The repository includes a conda environment file `environment.yml` with the
 specification for all development requirements to build and test the project.
```

Comment:
> The repository includes a conda environment file `environment.yml` with the
specification for all development requirements to build and test the project.

Actually I feel this sentence is very clear that all dependencies are listed in the environment.yml and people who want to build and test should read the environment.yml file, rather than listing an incompleted list here.

Reply:
A compromise could be to add more sections (via comments) to `environment.yml` that specify which development dependencies are required for building the docs, running the tests, and/or formatting the code (inspiration from [numpy](https://github.com/numpy/numpy/blob/main/environment.yml) and [pandas](https://github.com/pandas-dev/pandas/blob/main/environment.yml) environment files).

Reply:
> A compromise could be to add more sections (via comments) to `environment.yml` that specify which development dependencies are required for building the docs, running the tests, and/or formatting the code (inspiration from [numpy](https://github.com/numpy/numpy/blob/main/environment.yml) and [pandas](https://github.com/pandas-dev/pandas/blob/main/environment.yml) environment files).

Yes, that sounds like a good idea actually.

So do we not want to list some of the dev dependencies here in contributing.md? I know it seems redundant, but there are some people who might be from a non-Anaconda/pure-Python-pip sort of world that don't really use that environment.yml file. Listing things like git/dvc/pytest/sphinx here up front will at least give people an idea on what framework PyGMT uses for testing and building documentation.

Reply:
> > A compromise could be to add more sections (via comments) to `environment.yml` that specify which development dependencies are required for building the docs, running the tests, and/or formatting the code (inspiration from [numpy](https://github.com/numpy/numpy/blob/main/environment.yml?rgh-link-date=2022-03-04T16%3A39%3A12Z) and [pandas](https://github.com/pandas-dev/pandas/blob/main/environment.yml?rgh-link-date=2022-03-04T16%3A39%3A12Z) environment files).
> 
> Yes, that sounds like a good idea actually.

Yes, I agree.

> Listing things like git/dvc/pytest/sphinx here up front will at least give people an idea on what framework PyGMT uses for testing and building documentation.

Fair point.

Reply:
Ok, `environment.yml` has been reorganized into subsections (general, linters, testing, documentation) in ab8ac74178d077cdcb80e86a26d95bd273373267.

Reply:
Should this paragraph also include a short sentence on how to install dependencies with a .yml file? Since it's different than the typical `pip install -r requirements.txt`, it would cut down on confusion of how to use `environment.yml` to install dependencies.

Reply:
Good point. There's actually a `conda env create` instruction just a few sentences below. I've just added a commit (c2ae7dd44d2efa48f34873348cca73920761e14b) to make it more explicit (`conda env create --file environment.yml`).